src/main/java/htsjdk/samtools/util/OverlapDetector.java

```diff
@@ -53,74 +54,140 @@ public OverlapDetector(int lhsBuffer, int rhsBuffer) {
-        {
-            Set<T> alreadyThere = tree.put(start, end, objects);
-            if (alreadyThere != null)
-            {
+        if (start <= end) {  // Don't put in sequences that have no overlappable bases
+            final Set<T> alreadyThere = tree.put(start, end, objects);
+            if (alreadyThere != null) {
                 alreadyThere.add(object);
                 tree.put(start, end, alreadyThere);
             }
         }
     }
 
-    /** Adds all items to the overlap detector. */
-    public void addAll(List<T> objects, List<Interval> intervals) {
+    /**
+     * Adds all items to the overlap detector.
+     *
+     * The order of the lists matters only in the sense that it needs to be the same for the intervals
+     * and the corresponding objects.
+     */
+    public void addAll(final List<T> objects, final List<? extends Locatable> intervals) {
+        if (objects == null){
+            throw new IllegalArgumentException("null objects");
+        }
+        if (intervals == null){
+            throw new IllegalArgumentException("null intervals");
+        }
         if (objects.size() != intervals.size()) {
-            throw new IllegalArgumentException("Objects and intervals must be the same size.");
+            throw new IllegalArgumentException("Objects and intervals must be the same size but were " + objects.size() + " and " + intervals.size());
         }
 
         for (int i=0; i<objects.size(); ++i) {
             addLhs(objects.get(i), intervals.get(i));
         }
     }
 
-    /** Gets all the objects that could be returned by the overlap detector. */
-    public Collection<T> getAll() {
-        Collection<T> all = new HashSet<T>();
-        for (IntervalTree<Set<T>> tree : this.cache.values()) {
+    /**
+     * Gets all the objects that could be returned by the overlap detector.
+     */
+    public Set<T> getAll() {
+        final Set<T> all = new HashSet<>();
+        for (final IntervalTree<Set<T>> tree : this.cache.values()) {
             for (IntervalTree.Node<Set<T>> node : tree) {
                 all.addAll(node.getValue());
             }
         }
-
         return all;
     }
 
-    /** Gets the collection of objects that overlap the provided mapping. */
-    public Collection<T> getOverlaps(Interval rhs)  {
-        Collection<T> matches = new ArrayList<T>();
+    /**
+     * Returns true iff the given locatable overlaps any locatable in this detector.
+     *
+     * This is a performance shortcut API functionally equivalent to:
+     * <pre>{@code
+     *      ! getOverlaps(locatable).isEmpty()
+     * }</pre>
+     */
+    public boolean overlapsAny(final Locatable locatable) {
+        if (locatable == null){
+            throw new IllegalArgumentException("null locatable");
+        }
+        final String seqId = locatable.getContig();
+        final IntervalTree<Set<T>> tree = this.cache.get(seqId);
+        if (tree == null){
```

Comment:
IMO this space is useless waste and decreases readability. But fine, i will follow this convention